marklogic-client-api-functionaltests/src/test/java/com/marklogic/client/fastfunctest/TestSearchPageLength.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright © 2024 MarkLogic Corporation. All Rights Reserved.
+ */
+
+package com.marklogic.client.fastfunctest;
+
+import com.marklogic.client.impl.OkHttpServices;
+import com.marklogic.client.impl.RESTServices;
+import com.marklogic.client.io.*;
+import com.marklogic.client.query.*;
+import org.custommonkey.xmlunit.exceptions.XpathException;
+import org.junit.jupiter.api.*;
+import org.w3c.dom.Document;
+
+import static org.custommonkey.xmlunit.XMLAssert.*;
+
+
+class TestSearchPageLength extends AbstractFunctionalTest {
+
+	static final QueryManager queryMgr = client.newQueryManager();
+	StringQueryDefinition qd = queryMgr.newStringDefinition();
+
+	@Test
+	void testSearchPageLengthZero() throws XpathException {
+		queryMgr.setPageLength(0);
+		qd.setCriteria("Foo");
+		DOMHandle resultsHandle = new DOMHandle();
```

Comment:
Use a SearchHandle instead - that avoids awkward XPath expressions. And it lets you add a nice assertion message - i.e. "The server allows for a zero page length, which results in no results being returned." And also assert that no results are actually returned - i.e. `assertEquals(0, handle.getMatchResults().length)` (I'm curious if getMatchResults() returns null there or not).